test/wrappers_test.js

```diff
@@ -146,3 +149,75 @@ describe('transform-node', function() {
     chai.expect(spy).to.have.been.calledWith();
   });
 });
+
+describe('output-wrapper', function() {
+  let output;
+
+  beforeEach(() => {
+    fs.mkdirSync('tmp');
+    output = new OutputWrapper('tmp');
+  });
+
+  afterEach(() => {
+    rimraf.sync('tmp');
+  });
+
+  it('should write to given location', function() {
+    output.writeFileSync('test.md', 'test');
+    let content = fs.readFileSync('tmp/test.md', 'UTF-8');
+    expect(content).to.be.equal('test');
+  });
+
+  it('should throw if given location is empty/false', function() {
+    expect(() => output.writeFileSync(null, 'test')).to.throw('relativePath can not be null');
+  });
+
+  describe('should write to given location provided filemeta', function() {
+    it('getDestinationPath', function() {
+      let options = {
+        fileMeta: {
+          getDestinationPath(relativePath) {
```

Comment:
This is a very bizarre API, why would you call `.writeFileSync` with an argument _different_ than where you wanted to write the file?!

Reply:
The thinking behind API was if a `broccoli-plugin` does a mergeTree/funnel just for renaming folder or filename, this api can help the plugin to avoid merging/funneling. `getDestinationPath` will provide a handler to test and change path and `prefix` will just prepend the path. It has a good use case in `broccoli-persistent-filter` where plugins do multiple merges and funneling before the final tree is sent to `broccoli-persistent-filter`. But not in general broccoli of use case.